README.md
# React Auth Shared

[![](https://img.shields.io/circleci/project/github/etn-ccis/blui-react-auth-shared/master.svg?style=flat)](https://circleci.com/gh/etn-ccis/blui-react-auth-shared/tree/master) ![npm (scoped)](https://img.shields.io/npm/v/@brightlayer-ui/react-auth-shared) [![codecov](https://codecov.io/gh/etn-ccis/blui-react-auth-shared/branch/master/graph/badge.svg?token=BKDXMXPSDI)](https://codecov.io/gh/etn-ccis/blui-react-auth-shared)

The React Auth Shared package provides extracts the common logic shared by [@brightlayer-ui/react-native-auth-workflow](https://www.npmjs.com/package/@brightlayer-ui/react-native-auth-workflow) and [@brightlayer-ui/react-auth-workflow](https://www.npmjs.com/package/@brightlayer-ui/react-auth-workflow). This package must be used in combination with workflow screen definitions (either by using one of the packages above, or by building your own and integrating them yourself).

# Installation

To install the latest version of this package, run:

```shell
npm install --save @brightlayer-ui/react-auth-shared
// or
yarn add @brightlayer-ui/react-auth-shared
```

You will also need to install all of the peer dependencies. You can get the latest version of them all by running:

```
npm install --save date-fns@^2.19.0 i18next@^20.1.0 react@^17.0.0 react-i18next@^11.8.11
// or
yarn add date-fns@^2.19.0 i18next@^20.1.0 react@^17.0.0 react-i18next@^11.8.11
```

# Usage

We recommend you use the [@brightlayer-ui/react-native-auth-workflow](https://www.npmjs.com/package/@brightlayer-ui/react-native-auth-workflow) or [@brightlayer-ui/react-auth-workflow](https://www.npmjs.com/package/@brightlayer-ui/react-auth-workflow) packages, which use this shared logic under the hood. If you want to build your own screens and connect to this shared log, please contact us.

# Documentation

More information about exported objects and functions can found in the [API](https://github.com/etn-ccis/blui-react-auth-shared/tree/master/docs/API.md) documentation.

# Contributors

To work on this package as a contributor, first clone down the repository:

```shell
git clone https://github.com/etn-ccis/blui-react-auth-shared
cd react-auth-shared
```

You can build the library by running:

```shell
yarn build
```

You can run the Lint checks, prettier formatter, typescript validator, and unit tests by running:

```shell
yarn validate
```

You can update the auto-generated licenses.md file by running:

```shell
yarn generate:licenses
```


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## v3.8.0 (October 12, 2023)

### Added

-   Ability to hide the CreatePassword screen
-   Ability to hide the login form in the Login screen.

### Changed

-   Changed resend email button's text label on the email verification screen.

## v3.7.4 (November 30, 2022)

### Changed

-   Update `AccountDetailsFormProps` `ref` type definition to be `React.RefObject<any>`.

## v3.7.3 (October 28, 2022)

### Fixed

-   Refactor functions that use regex named capture groups to fix conflicts with hermes engine.

## v3.7.2 (September 7, 2022)

### Added

-   Add `USERNAME_ENTRY_ERROR` translations.
-   Add `USERNAME_REGEX` constant for validating username entries.
-   Add optional `ref` prop to `AccountDetailsFormProps`.

## v3.7.1 (August 4, 2022)

### Added

-   Add `VERIFICATION_CODE_PROMPT` translations.

### Fixed

-   Fix Portuguese translation issues ([#74](https://github.com/etn-ccis/blui-react-auth-shared/issues/74)).

## v3.7.0 (April 26, 2022)

### Added

-   `disablePagerAnimation` prop to `AuthUIContextProvider` to disable pager animations.

## v3.6.2 (April 14, 2022)

### Added

-   Support for React 18.

## v3.6.1 (April 1, 2022)

### Fixed

-   Fixed non-bold text on Forgot Password screen when using Portuguese translations ([#59](https://github.com/etn-ccis/blui-react-auth-shared/issues/59)).

## v3.6.0 (December 9, 2021)

### Added

-   Support for Portuguese translations.
-   Type checking for errors.

## v3.5.0 (November 11, 2021)

### Changed

-   Changed package namespace from `@pxblue` to `@brightlayer-ui`.

## Package Migration Notice

Previous versions listed after this indicator refer to our deprecated `@pxblue` packages.

---

## v3.5.0 (August 13, 2021)

### Added

-   `loginErrorDisplayConfig` prop to `AuthUIContextProvider` to customize how errors are displayed on the login screen.

## v3.4.0 (July 21, 2021)

### Added

-   `registrationConfig` prop to `AuthUIContextProvider` to customize the character limits for first and last name text fields in the registration workflow.

## v3.3.0 (June 22, 2021)

### Added

-   New type for LoginType.
-   New translations for "username".

## v3.2.1 (May 18, 2021)

### Fixed

-   Corrected some Spanish translations.
-   Missing space in support strings for spanish translation file.

## v3.2.0 (April 29, 2021)

### Added

-   Support for Spanish and Chinese translations.
-   Strong typings for translation files.
-   Updated type definition for `registrationSuccessScreen`.

### Fixed

-   Incorrect key name in French translation file.

## v3.1.0 (April 26, 2021)

### Added

-   Updated type definitions to allow for a custom success screen in the Registration flow.

## v3.0.0 (March 29, 2021)

### Added

-   Updated type definitions to allow for further customization of the elements on the Login screen.
-   Updated type definitions to allow for custom forms / data collection in the Registration flow.
-   Peer dependency support for React 17.

### Removed

-   Phone number is no longer part of the default registration properties (you must add it to your custom forms to collect this data)
-   Unused `title` prop from `AuthUIContextProvider`.
-   Highcharts formatting functions — these have been moved to [@brightlayer-ui/highcharts](https://www.npmjs.com/package/@brightlayer-ui/highcharts).

## v2.1.0 (November 10, 2020)

### Added

-   `phoneContactLink` to `AuthUIContextProviderProps` to allow specification of tel URI details while keeping `phoneContact` for human-readable UI display of the same number.

## v2.0.1 (August 28, 2020)

### Added

-   New translation strings for React applications using `<Trans>` component.

## v2.0.0 (August 10, 2020)

### Removed

-   default export from i18n utilities
    -   This will be implemented in [@brightlayer-ui/react-native-auth-workflow](https://www.npmjs.com/package/@brightlayer-ui/react-native-auth-workflow) and [@brightlayer-ui/react-auth-workflow](https://www.npmjs.com/package/@brightlayer-ui/react-auth-workflow) separately.
-   DismissKeyboardView component
    -   Moved to @brightlayer-ui/react-native-auth-workflow

### Changed

-   Type for `projectImage` has changed from `ImageSourcePropType` to `number | string` to support React Native _and_ React projects.

## v1.0.1 (July 16, 2020)

### Changed

-   Moved re-usable react logic from @brightlayer-ui/react-native-auth-workflow to @brightlayer-ui/react-auth-shared.


src/data/translations/english.ts
import { LanguageFile } from './types';

const resources: LanguageFile = {
    translation: {
        ACTIONS: {
            FINISH: 'Finish',
            NEXT: 'Next',
            BACK: 'Back',
            CREATE_ACCOUNT: 'Create Account',
            OKAY: 'Okay',
            CANCEL: 'Cancel',
            CONTINUE: 'Continue',
            DONE: 'Done',
            LOG_IN: 'Log In',
            LOG_OUT: 'Log Out',
            CLICK_BUTTON: 'Click the button',
            UPDATE_REDUX: 'Click the button to update the redux store value!',
            CHANGE_LANGUAGE: 'Change Language here!',
            GO_HOME: 'Go to Home',
            GO_TEST: 'Go to Test Page',
            RESEND: 'Send Again',
            UPDATE: 'Update',
            REMEMBER: 'Remember Me',
        },
        LABELS: {
            EMAIL: 'Email Address',
            USERNAME: 'Username',
            PASSWORD: 'Password',
            CURRENT_PASSWORD: 'Current Password',
            NEW_PASSWORD: 'New Password',
            OPTIONAL: 'Optional',
            FORGOT_PASSWORD: 'Forgot your password?',
            NEED_ACCOUNT: 'Need an account?',
            VIEW_ALL_EVENTS: 'View All {{count}} Events',
        },
        MESSAGES: {
            EMAIL_SENT: 'Email Sent',
            WELCOME: 'Welcome',
            WELCOME_PROJECT: 'Welcome to {{project}}',
            LOGIN_MESSAGE: "You are 'logged' in",
            CONGRATS: 'Congratulations!',
            CONTACT: 'Contact an Eaton Support Representative',
            ERROR: 'Error!',
            EMAIL_ENTRY_ERROR: 'Please enter a valid email',
            USERNAME_ENTRY_ERROR: 'Please enter a valid username',
            SUCCESS: 'Success',
            FAILURE: 'Failure',
            LOADING: 'Loading...',
            REQUEST_ERROR: 'Sorry, there was a problem sending your request.',
        },
        REGISTRATION: {
            EULA: {
                LOADING: 'Loading End User License Agreement...',
                AGREE_TERMS: 'I have read and agree to the Terms & Conditions',
            },
            STEPS: {
                CREATE_ACCOUNT: 'Create an Account',
                VERIFY_EMAIL: 'Verify Email',
                LICENSE: 'License Agreement',
                PASSWORD: 'Create Password',
                ACCOUNT_DETAILS: 'Account Details',
                COMPLETE: 'Account Created!',
            },
            INSTRUCTIONS: {
                ACCOUNT_DETAILS: 'Enter your details below to complete account creation.',
            },
            SUCCESS_MESSAGE:
                'Your account has been successfully created with the email <b>{{email}}</b>.\n\nYour account has already been added to the <b>{{organization}}</b> organization.\n\nPress continue below to finish.',
            SUCCESS_MESSAGE_ALT:
                'Your account has been successfully created with the email <1>{{email}}</1>.\n\nYour account has already been added to the <3>{{organization}}</3> organization.\n\nPress continue below to finish.',
            SUCCESS_EXISTING:
                'Your account has been successfully created. Please log in with your Eaton account email and password.',
            FAILURE_MESSAGE: 'We were unable to complete your registration. Press continue below to finish.',
            UNKNOWN_EMAIL: 'Unknown Email',
            UNKNOWN_ORGANIZATION: 'Unknown Organization',
        },
        SELF_REGISTRATION: {
            INSTRUCTIONS: `To register for an Eaton account, enter the required information below. You will need to verify your email address to continue.`,
            VERIFY_EMAIL: {
                MESSAGE:
                    'A verification code has been sent to the email address you provided. Click the link or enter the code below to continue. This code is valid for 30 minutes.',
                RESEND: 'Resend Verification Email',
                VERIFICATION_CODE_PROMPT: "Didn't receive email?",
                VERIFICATION: 'Verification Code',
            },
        },
        FORGOT_PASSWORD: {
            ERROR: 'Could not reset your password at this time.',
            INSTRUCTIONS:
                'Please enter the account email associated with the account.\n\n' +
                'If this email has an account with Eaton, you will receive a response within <b>one business day</b>.\n\n' +
                'For urgent account issues, please call {{phone}}.',
            INSTRUCTIONS_ALT:
                'Please enter the account email associated with the account.<br/><br/>' +
                'If this email has an account with Eaton, you will receive a response within <1>one business day</1>.<br/><br/>' +
                'For urgent account issues, please call <4>{{phone}}</4>.',
            RESET_CODE_ERROR: 'There was an error with your reset code. ',
            LINK_SENT: 'A link to reset your password has been sent to <b>{{email}}</b>.',
            LINK_SENT_ALT: 'A link to reset your password has been sent to <1>{{email}}</1>.',
        },
        LOGIN: {
            INCORRECT_CREDENTIALS: 'Incorrect Email or Password',
            INVALID_CREDENTIALS: 'Your username/password combination is not recognized.',
            GENERIC_ERROR: 'Your request could not be processed at this time.',
        },
        FORMS: {
            FIRST_NAME: 'First Name',
            LAST_NAME: 'Last Name',
            PHONE_NUMBER: 'Phone Number',
            PASSWORD: 'Password',
            CONFIRM_PASSWORD: 'Confirm Password',
            PASS_MATCH_ERROR: 'Passwords do not match',
            TOGGLE_PASSWORD_VISIBILITY: 'Toggle password visibility',
            RESET_PASSWORD: 'Reset Password',
        },
        PASSWORD_REQUIREMENTS: {
            LENGTH: '8-16 Characters',
            NUMBERS: 'One number',
            UPPER: 'One uppercase letter',
            LOWER: 'One lowercase letter',
            SPECIAL: 'One special character',
        },
        PASSWORD_RESET: {
            SUCCESS_MESSAGE: 'Your password was successfully reset.',
            FAILURE_MESSAGE: 'Your password was unable to be reset. Please try again later.',
        },
        USER_SETTINGS: {
            NAME: 'Name',
            EMAIL: 'Email',
            PHONE_NUMBER: 'Phone Number',
            PASSWORD: 'Password',
            EMAIL_NOTIFICATION: 'Email Notification',
            ENABLED: 'Enabled',
            ORGANIZATION: 'Organization',
            ORGANIZATION_NAME: 'Organization Name',
            ADDRESS: 'Address',
            CHANGE_PASSWORD: 'Change Password',
            ACCOUNT: 'Account',
        },
        COUNTER: 'The value is: {{count}}',
        HEADER: {
            FORGOT_PASSWORD: 'Forgot Password',
        },
        ERROR_MESSAGES: {
            '2002': 'User registration link is already redeemed.',
            '9003': 'Requested operation cannot be performed, please contact your administrator',
        },
        CHANGE_PASSWORD: {
            PASSWORD_CHANGED: 'Password Changed',
            PASSWORD: 'Change Password',
            SUCCESS_MESSAGE:
                "Your password was successfully updated! To ensure your account's security, you will need to log in to the application with your updated credentials.",
            EMAIL_CONFIRM_MESSAGE: 'We have sent a confirmation email to <b>{{email}}</b>',
            PASSWORD_INFO:
                'Please select a password. Make sure that your password meets the necessary complexity requirements outlined below.',
            OLD_PASSWORD: 'Old Password',
            ERROR_MESSAGE:
                'Your information did not match our records. Please re- enter your information to try again.',
            PROBLEM_OCCURRED: 'A problem occurred:',
            CONFIRM_NEW_PASSWORD: 'Confirm New Password',
            CANCEL: 'Cancel',
            UPDATE: 'Update',
        },
        SETTINGS: {
            TITLE: 'Settings',
        },
        LEGAL: {
            TITLE: 'Legal',
            TERMSANDCONDITIONS: 'Terms and Conditions',
            EULA: 'End User License Agreement',
            OPENSOURCELICENSES: 'Open Source Licenses',
        },
        USER_MENU: {
            LOG_OUT: 'Log Out',
            CONTACT_US: 'Contact Us',
            ACCOUNT_SETTING: 'Account Settings',
        },
        CONTACT_SUPPORT: {
            GENERAL_QUESTIONS: 'General Questions',
            SUPPORT_MESSAGE: 'For questions, feedback, or support please email us at ',
            EMERGENCY_SUPPORT: 'Emergency Support',
            TECHNICAL_ASSISTANCE: 'For technical support, please call ',
        },
    },
};
export default resources;
